tests.py

```diff
@@ -0,0 +1,212 @@
+import cufflinks as cf
+import pandas as pd
+
+import unittest
+
+
+class TestIPlot(unittest.TestCase):
+    def setUp(self):
+        self.df = pd.DataFrame(dict(x=[1, 2, 3], y=[4, 2, 1], c=[3, 1, 5]))
+
+    def _iplot(self, df, **kwargs):
+        return df.iplot(asFigure=True, **kwargs)
+
+    @unittest.skip("no asFigure method")
+    def test_scatter_matrix(self):
+        self.df.scatter_matrix(asFigure=True)
+
+    @unittest.skip("example from docs, but this doesnt work")
+    def test_irregular_subplots(self):
+        df = cf.datagen.bubble(10, 50, mode='stocks')
+        figs = cf.figures(df, [
+            dict(kind='histogram', keys='x', color='blue'),
+            dict(kind='scatter', mode='markers', x='x', y='y', size=5),
+            dict(kind='scatter', mode='markers', x='x', y='y',
+                 size=5, color='teal')])
+        figs.append(cf.datagen.lines(1).figure(bestfit=True, colors=['blue'],
+                                               bestfit_colors=['pink']))
+        base_layout = cf.tools.get_base_layout(figs)
+        sp = cf.subplots(figs, shape=(3, 2), base_layout=base_layout,
+                         vertical_spacing=.15, horizontal_spacing=.03,
+                         specs=[[{'rowspan': 2}, {}], [None, {}],
+                                [{'colspan': 2}, None]],
+                         subplot_titles=['Histogram', 'Scatter 1',
+                                         'Scatter 2', 'Bestfit Line'])
+        sp['layout'].update(showlegend=False)
+
+        cf.iplot(sp, asFigure=True)
+
+
+def bar_input_argument_tests():
+    options = {
+        'kind': ['bar', 'barh'],
+        'barmode': ['stack', 'overlay', 'group'],
+        'bargap': [0.1],
+        'subplots': [True]
+    }
+
+    def bar_test(self, **kwargs):
+        self._iplot(self.df, **kwargs)
+
+    _generate_tests(TestIPlot, bar_test, 'bar', options)
```

Comment:
But it's a nice way to verify and declare all of the valid input argument combinations.